src/data/biome-utils.ts

```diff
@@ -49,3 +49,224 @@ export interface BiomeTierTrainerPools {
+  BiomeId.DESERT,
+  BiomeId.POWER_PLANT,
+  BiomeId.CHARGESTONE_CAVE,
+  BiomeId.DOJO,
+  BiomeId.RUINS,
+  BiomeId.CONSTRUCTION_SITE,
+];
+export const saltedCreamAlcremieBiomes = [
+  BiomeId.MOUNTAIN,
+  BiomeId.CAVE,
+  BiomeId.ICE_CAVE,
+  BiomeId.FAIRY_CAVE,
+  BiomeId.SNOWY_FOREST,
+];
+export const rubySwirlAlcremieBiomes = [BiomeId.WASTELAND, BiomeId.LABORATORY];
+export const caramelSwirlAlcremieBiomes = [BiomeId.TEMPLE, BiomeId.ISLAND];
+export const rainbowSwirlAlcremieBiomes = [BiomeId.SPACE, BiomeId.ABYSS, BiomeId.END];
+
+// #region ME biome mappings
+export const EXTREME_ENCOUNTER_BIOMES = [
+  BiomeId.SEA,
+  BiomeId.SEABED,
+  BiomeId.BADLANDS,
+  BiomeId.DESERT,
+  BiomeId.ICE_CAVE,
+  BiomeId.VOLCANO,
+  BiomeId.WASTELAND,
+  BiomeId.ABYSS,
+  BiomeId.SPACE,
+  BiomeId.END,
+];
+
+export const NON_EXTREME_ENCOUNTER_BIOMES = [
+  BiomeId.TOWN,
+  BiomeId.PLAINS,
+  BiomeId.GRASS,
+  BiomeId.TALL_GRASS,
+  BiomeId.METROPOLIS,
+  BiomeId.FOREST,
+  BiomeId.SWAMP,
+  BiomeId.BEACH,
+  BiomeId.LAKE,
+  BiomeId.MOUNTAIN,
+  BiomeId.CAVE,
+  BiomeId.MEADOW,
+  BiomeId.POWER_PLANT,
+  BiomeId.GRAVEYARD,
+  BiomeId.DOJO,
+  BiomeId.FACTORY,
+  BiomeId.RUINS,
+  BiomeId.CONSTRUCTION_SITE,
+  BiomeId.JUNGLE,
+  BiomeId.FAIRY_CAVE,
+  BiomeId.TEMPLE,
+  BiomeId.SLUM,
+  BiomeId.SNOWY_FOREST,
+  BiomeId.ISLAND,
+  BiomeId.LABORATORY,
+];
+/**
+ * Places where you could very reasonably expect to encounter a single human
+ *
+ * Diff from NON_EXTREME_ENCOUNTER_BIOMES:
+ * + BADLANDS
+ * + DESERT
+ * + ICE_CAVE
+ */
+
+export const HUMAN_TRANSITABLE_BIOMES = [
+  BiomeId.TOWN,
+  BiomeId.PLAINS,
+  BiomeId.GRASS,
+  BiomeId.TALL_GRASS,
+  BiomeId.METROPOLIS,
+  BiomeId.FOREST,
+  BiomeId.SWAMP,
+  BiomeId.BEACH,
+  BiomeId.LAKE,
+  BiomeId.MOUNTAIN,
+  BiomeId.BADLANDS,
+  BiomeId.CAVE,
+  BiomeId.CHARGESTONE_CAVE,
+  BiomeId.STEAM_VENT,
+  BiomeId.DESERT,
+  BiomeId.ICE_CAVE,
+  BiomeId.MEADOW,
+  BiomeId.POWER_PLANT,
+  BiomeId.GRAVEYARD,
+  BiomeId.DOJO,
+  BiomeId.FACTORY,
+  BiomeId.RUINS,
+  BiomeId.CONSTRUCTION_SITE,
+  BiomeId.JUNGLE,
+  BiomeId.FAIRY_CAVE,
+  BiomeId.TEMPLE,
+  BiomeId.SLUM,
+  BiomeId.SNOWY_FOREST,
+  BiomeId.ISLAND,
+  BiomeId.LABORATORY,
+];
+/**
+ * Places where you could expect a town or city, some form of large civilization
+ */
+
+export const CIVILIZATION_ENCOUNTER_BIOMES = [
+  BiomeId.TOWN,
+  BiomeId.PLAINS,
+  BiomeId.GRASS,
+  BiomeId.TALL_GRASS,
+  BiomeId.METROPOLIS,
+  BiomeId.BEACH,
+  BiomeId.LAKE,
+  BiomeId.MEADOW,
+  BiomeId.POWER_PLANT,
+  BiomeId.GRAVEYARD,
+  BiomeId.DOJO,
+  BiomeId.FACTORY,
+  BiomeId.CONSTRUCTION_SITE,
+  BiomeId.SLUM,
+  BiomeId.ISLAND,
+];
+
+// #region dancing-lesson ME
+
+// Fire form
+export const BAILE_STYLE_BIOMES = [
+  BiomeId.VOLCANO,
+  BiomeId.STEAM_VENT,
+  BiomeId.BEACH,
+  BiomeId.ISLAND,
+  BiomeId.WASTELAND,
+  BiomeId.MOUNTAIN,
+  BiomeId.BADLANDS,
+  BiomeId.DESERT,
+];
+
+// Electric form
+export const POM_POM_STYLE_BIOMES = [
```

Comment:
```suggestion
export const POM_POM_STYLE_BIOMES = [
  BiomeId.TOWN,
```
It turns out a bunch of biomes have been missing from the Dancing Lessons biome lists the entire time...

Reply:
The way theses are set up (if/else checking) I can see collapsing this into a switch actually